core/src/main/java/discord4j/core/event/dispatch/GuildDispatchHandlers.java

```diff
@@ -217,13 +217,14 @@ static Mono<MemberUpdateEvent> guildMemberUpdate(DispatchContext<GuildMemberUpda
         String currentPremiumSince = Possible.flatOpt(context.getDispatch().premiumSince()).orElse(null);
         Boolean currentPending = context.getDispatch().pending().toOptional().orElse(null);
         String communicationDisabledUntil = Possible.flatOpt(context.getDispatch().communicationDisabledUntil()).orElse(null);
+        AvatarDecoration avatarDecoration = Possible.flatOpt(context.getDispatch().avatarDecoration()).map(avatarDecorationData -> new AvatarDecoration(gateway, avatarDecorationData)).orElse(null);
```

Comment:
This will be used in a method returning an Optional afterwards so you can already keep it as an Optional here

Reply:
hmm yeah i make that mostly based in how the event make the same thing with another fields